packages/core/src/hub.ts

```diff
@@ -506,13 +507,7 @@ export class Hub implements HubInterface {
  * @deprecated Use the respective replacement method directly instead.
  */
 // eslint-disable-next-line deprecation/deprecation
-export function getCurrentHub(): HubInterface {
-  // Get main carrier (global for every environment)
-  const carrier = getMainCarrier();
-
-  const acs = getAsyncContextStrategy(carrier);
-  return acs.getCurrentHub() || getGlobalHub();
-}
```

Comment:
It's no longer necessary to read from the carrier here since acs doesn't have the `getCurrentHub` method anymore. This means that we can replace this entire implementation with the shim which in turn will again forward its respective calls to the `globalHub` (to be renamed).

Reply:
I think we need to remove `getGlobalHub` usage and make everything rely on the shim, though that does mean we need to refactor more of the return values in `getHubStackAsyncContextStrategy`

Reply:
yes, I talked about this with @mydea and that's basically what we should do. `getHubStackAsyncContextStrategy` has to stay in one way or the other since our fallback (i.e. browser) async context strategy is still a stack-based scope hierarchy. `getGlobalHub` currently returns a `Hub` class instance which we can probably cut down a lot to only handle the scope stack. At this point we can then also rename this to something like `globalScopeStackAsyncContextStrategy`.